src/main/java/com/shipmentEvents/demo/App.java

```diff
@@ -0,0 +1,102 @@
+package com.mycompany.app;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.JSONInput;
+import com.amazonaws.services.s3.model.JSONOutput;
+import com.amazonaws.services.s3.model.CompressionType;
+import com.amazonaws.services.s3.model.ExpressionType;
+import com.amazonaws.services.s3.model.InputSerialization;
+import com.amazonaws.services.s3.model.OutputSerialization;
+import com.amazonaws.services.s3.model.SelectObjectContentRequest;
+import com.amazonaws.services.s3.model.SelectObjectContentResult;
+import com.google.gson.*;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.sql.SQLOutput;
+import java.util.*;
+import java.io.InputStream;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.io.IOException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+
+
+public class App implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+    private static final String S3_BUCKET = "aws-tc-largeobjects";
+    private static final String S3_FILE = "DEV-AWS-MO-Building_2.0/my_json_lines.jsonl";
+    private static final Gson gson = new Gson();
+    private static final String QUERY_PARAM_NAME = "product_id";
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent event, Context context) {
+        String data = readData(event);
+        return generateResponse(data, event.getQueryStringParameters().get(QUERY_PARAM_NAME));
+    }
+
+    protected static String readData(APIGatewayProxyRequestEvent event) {
+        Map<String,String> queryParams = event.getQueryStringParameters();
+        SelectObjectContentRequest request = generateJSONRequest(queryParams);
+        return queryS3(request);
+    }
+
+    private static String queryS3(SelectObjectContentRequest request) {
+        final AtomicBoolean isResultComplete = new AtomicBoolean(false);
+        // Call S3 Select
+        SelectObjectContentResult result = s3Client.selectObjectContent(request);
+        JsonArray array = new JsonArray();
+
+        try {
+
+            InputStream resultInputStream = result.getPayload().getRecordsInputStream();
+
+            BufferedReader streamReader = new BufferedReader(new InputStreamReader(resultInputStream, "UTF-8"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *streamReader*. Execution paths that do not contain closure statements were detected. Ensure *streamReader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).